OrbitQt/TopDownWidget.cpp

```diff
@@ -374,3 +396,50 @@ QVariant TopDownWidget::HookedIdentityProxyModel::data(const QModelIndex& index,
       QStringLiteral("] ");
   return kDisplayHookedPrefix + data.toString();
 }
+
+void TopDownWidget::ProgressBarItemDelegate::paint(QPainter* painter,
+                                                   const QStyleOptionViewItem& option,
+                                                   const QModelIndex& index) const {
+  bool is_float = false;
+  float inclusive_percent = index.data(Qt::EditRole).toFloat(&is_float);
+  if (!is_float) {
+    QStyledItemDelegate::paint(painter, option, index);
+    return;
+  }
+
+  bool highlight =
+      index.data(HighlightCustomFilterSortFilterProxyModel::kMatchesCustomFilterRole).toBool();
+  if (option.state & QStyle::State_Selected) {
+    painter->fillRect(option.rect, option.palette.highlight());
+    // Don't highlight the progress bar text when the row is selected, for consistency with the
+    // other columns.
+    highlight = false;
+  }
+
+  QStyleOptionProgressBar option_progress_bar;
+  option_progress_bar.rect = option.rect;
+  option_progress_bar.minimum = 0;
+  option_progress_bar.maximum = 100;
+  option_progress_bar.progress = static_cast<int>(round(inclusive_percent));
+
+  option_progress_bar.palette.setColor(QPalette::Base,
+                                       option.palette.color(QPalette::Disabled, QPalette::Base));
+  const QColor bar_color =
+      QColor::fromHsv(option.palette.color(QPalette::Highlight).hue(),
+                      option.palette.color(QPalette::Highlight).saturation(),
+                      option.palette.color(QPalette::Highlight).value() * 2 / 3);
```

Comment:
maybe define the 2/3 as a constant?

Reply:
nit. would it be possible to cache or compute the color once for all progress bars?

Reply:
> maybe define the 2/3 as a constant?

Done (needs some casting to float and back now, `2 / 3` with `int`s is 0...).

Reply:
> nit. would it be possible to cache or compute the color once for all progress bars?

I guess, but conceptually I would like to avoid that as `const QStyleOptionViewItem& option` is passed every time (the style could theoretically change). The overhead should be negligible.